pug/edit_component.pug

```diff
@@ -150,191 +147,116 @@ block extrascripts
         }
         else
         {
-          if (obj[key] == null)
+          if (obj[key] === null)
           {
             delete obj[key];
           }
           
-          if (typeof obj[key] === 'string' && obj[key].length == 0)
+          if (typeof obj[key] === 'string' && obj[key].length === 0)
           {
             delete obj[key];
           }
         }
       });
       
-      console.log("removeEmpty", inobj, obj);
       return obj;
     }
-
-    var recent = null;
     
+    // Search the DB for any component of the same type as this submission that was most recently modified (created or edited) by this user
     function searchForRecentSubmission()
     {
       $('#loadRecentData').hide();
       
-      $.post("/json/search/component/" + (component.type || "") + "?limit=1",
-             {
-              "insertion.user.user_id": "!{user.user_id}",
-             })
+      $.post('/json/search/component/' + (component.formId || "") + "?limit=1", {"insertion.user.user_id": "!{user.user_id}"})
```

Comment:
There is a bug here relating to the field name change (`component.type' -> `component.formId`).  For some reason, it is not recognising the new field name for components that definitely do have it, and therefore the button to load the recent data from such components is not appearing.  The wierd thing is that it DOES appear for components with the old field name.  This is as far as I could trace it ... can't seem to find what happens after this point.